test/thread_pool.cpp

```diff
@@ -22,4 +24,28 @@ TEST(ThreadPool, should_execute_function_in_another_thread) {
   EXPECT_NE(tid, std::this_thread::get_id());
 }
 
+TEST(ThreadPool, processes_all_queued_tasks_when_destroyed) {
+  static constexpr size_t TASK_COUNT = 1;
```

Comment:
SCREAMING_CASE is often used by macros. I prefer to use same snake_case naming convention for static variables inside functions and for regular local vars. I add `g_` prefix for global variables and still use snake_case for them.